website/static/js/newAndNoteworthyPlugin.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * New and Noteworthy Projects
+ */
+
+var $ = require('jquery');
+var m = require('mithril');
+var $osf = require('js/osfHelpers');
+
+// CSS
+require('css/quick-project-search-plugin.css');
+
+// XHR config for apiserver connection
+var xhrconfig = function(xhr) {
+    xhr.withCredentials = true;
+};
+
+
+var newAndNoteworthy = {
+    controller: function() {
+        var self = this;
+        self.newAndNoteworthyNodes = m.prop([]);
+        self.popularNodes = m.prop([]);
+        self.contributorsMapping = {};
+
+        // Load new and noteworthy nodes
+        var newAndNoteworthyUrl = $osf.apiV2Url('nodes/' + window.contextVars.newAndNoteworthy + '/node_links/', {});
+        var newAndNoteworthyPromise = m.request({method: 'GET', url: newAndNoteworthyUrl, config: xhrconfig});
+        newAndNoteworthyPromise.then(function(result){
+            var numPopular = Math.min(result.data.length - 1, 4);
+            for (var l=0; l <= numPopular; l++) {
+                self.newAndNoteworthyNodes().push(result.data[l]);
+                self.fetchContributors(result.data[l]);
+                  }
+            return newAndNoteworthyPromise;
```

Comment:
I don't think this is the appropriate way to return the promise. You can turn the loading noteworthy nodes into a function which returns the promise